Description: Add an "authenticate only" option
 This gives an init.d implementation a way to establish auth
 without having to execute the proxy.
From: David Steele <dsteele@gmail.com>

--- a/cloudprint/cloudprint.py
+++ b/cloudprint/cloudprint.py
@@ -381,14 +381,16 @@
     print '-a account_file\t: path to google account ident data (default ~/.cloudprintauth)'
     print '\t\t account_file format:\t <Google username>'
     print '\t\t\t\t\t <Google password>'
+    print '-c\t\t: establish and store login credentials, then exit'
     print '-h\t\t: display this help'
 
 def main():
-    opts, args = getopt.getopt(sys.argv[1:], 'dlhp:a:')
+    opts, args = getopt.getopt(sys.argv[1:], 'dlhp:a:c')
     daemon = False
     logout = False
     pidfile = None
     authfile = None
+    authonly = False
     saslauthfile = None
     for o, a in opts:
         if o == '-d':
@@ -400,6 +402,8 @@
         elif o == '-a':
             authfile = a
             saslauthfile = authfile+'.sasl'
+        elif o == '-c':
+            authonly = True
         elif o =='-h':
             usage()
             sys.exit()
@@ -436,8 +440,8 @@
                 account_file.close()
 
             else:
-                LOGGER.info('Unable to find account file')
-                return
+                cpp.username = raw_input('Google username: ')
+                cpp.password = getpass.getpass()
         else:
           cpp.username = raw_input('Google username: ')
           cpp.password = getpass.getpass()
@@ -457,6 +461,9 @@
             #reset the stored auth
             cpp.set_auth('')
 
+    if authonly:
+        sys.exit(0)
+
     printers = cpp.get_printers()
 
     if daemon:
